httptransport/notificationshandler.go

```diff
@@ -149,9 +146,8 @@ func (h *NotifHandler) Get(w http.ResponseWriter, r *http.Request) {
 		Notifications: notifications,
 	}
 
-	err = json.NewEncoder(w).Encode(&response)
-	if err != nil {
-		log.Error().Err(err).Msg("failed to serialize response")
-	}
-	return
+	defer writerError(w, &err)()
```

Comment:
Still confused about whether this is too useful, how is anything going to know to check a trailer?

Reply:
It's called out in the headers.

I waffle on how useful reporting it via the trailer is, but this is at least centralizing handling the writer errors in one spot, one way.